cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/templates/etl/batch/ETLBatchTemplate.java

```diff
@@ -86,9 +95,23 @@ public void configureAdapter(String adapterName, ETLBatchConfig etlBatchConfig,
 
   @Override
   public void configure(ApplicationConfigurer configurer, ApplicationContext context) {
-    configurer.setName("etlbatch");
-    configurer.setDescription("Batch Extract-Transform-Load (ETL) Adapter");
-    configurer.addMapReduce(new ETLMapReduce());
-    configurer.addWorkflow(new ETLWorkflow());
+    Properties prop = new Properties();
```

Comment:
I think you can move this logic into `ETLTemplate`. Something like `getAppName`.